From a8f91ef5a0bb181268f881f4f837739215da3745 Mon Sep 17 00:00:00 2001
From: Chaitanya Koparkar <ckoparkar@gmail.com>
Date: Fri, 11 Sep 2020 11:10:04 -0400
Subject: [PATCH] Patch dune.2.7.1 for multicore

---
 bootstrap.ml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/bootstrap.ml b/bootstrap.ml
index e36edd33..9ee378c0 100644
--- a/bootstrap.ml
+++ b/bootstrap.ml
@@ -65,9 +65,9 @@ let read_file fn =
   s
 
 let () =
-  let v = Scanf.sscanf Sys.ocaml_version "%d.%d" (fun a b -> (a, b)) in
+  let v, p = Scanf.sscanf Sys.ocaml_version "%d.%d%s@+%s" (fun a b c d -> (a, b), d) in
   let compiler, which =
-    if v >= min_supported_natively then
+    if v >= min_supported_natively && p <> "multicore" then
       ("ocamlc", None)
     else
       let compiler = "ocamlfind -toolchain secondary ocamlc" in
-- 
2.17.1

