From 7441d4cf95f3cfce844c606c2914af8e5b687340 Mon Sep 17 00:00:00 2001
From: David Allsopp <david.allsopp@metastack.com>
Date: Fri, 29 May 2020 07:35:35 +0100
Subject: [PATCH] Use secondary compiler for multicore

---
 bootstrap.ml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/bootstrap.ml b/bootstrap.ml
index 242487d1..835e49ed 100644
--- a/bootstrap.ml
+++ b/bootstrap.ml
@@ -65,9 +65,9 @@ let read_file fn =
   s
 
 let () =
-  let v = Scanf.sscanf Sys.ocaml_version "%d.%d" (fun a b -> (a, b)) in
+  let v, p = Scanf.sscanf Sys.ocaml_version "%d.%d%s@+%s" (fun a b c d -> (a, b), d) in
   let compiler, which =
-    if v >= min_supported_natively then
+    if v >= min_supported_natively && p <> "multicore" then
       ("ocamlc", None)
     else
       let compiler = "ocamlfind -toolchain secondary ocamlc" in
-- 
2.17.1

